File: tools/packages/fetch-client/packages_mjs/fetch-client/src/core/request.ts
/**
 * Request builder helper.
 */
import { HttpMethod, RequestOptions } from '../types.js';

export class RequestBuilder {
    private options: RequestOptions;

    constructor(url: string = '', method: HttpMethod = 'GET') {
        this.options = {
            url,
            method,
            headers: {},
            query: {}
        };
    }

    url(url: string): this {
        this.options.url = url;
        return this;
    }

    method(method: HttpMethod): this {
        this.options.method = method;
        return this;
    }

    header(key: string, value: string): this {
        this.options.headers = this.options.headers || {};
        this.options.headers[key] = value;
        return this;
    }

    headers(headers: Record<string, string>): this {
        this.options.headers = { ...this.options.headers, ...headers };
        return this;
    }

    param(key: string, value: any): this {
        this.options.query = this.options.query || {};
        this.options.query[key] = value;
        return this;
    }

    params(params: Record<string, any>): this {
        this.options.query = { ...this.options.query, ...params };
        return this;
    }

    json(data: any): this {
        this.options.json = data;
        return this;
    }

    /**
     * Set raw body.
     */
    body(data: string | Buffer | Uint8Array | null): this {
        this.options.body = data;
        return this;
    }

    timeout(timeout: number): this {
        this.options.timeout = timeout;
        return this;
    }

    build(): RequestOptions {
        return this.options;
    }
}

File: packages_py/fetch_client/src/fetch_client/core/request.py
"""
Request builder helper.
"""
from typing import Any, Dict, Optional, Union
from ..types import HttpMethod, RequestOptions

class RequestBuilder:
    """Fluent builder for RequestOptions."""
    
    def __init__(self, url: str = "", method: HttpMethod = "GET"):
        self._options: RequestOptions = {
            "url": url,
            "method": method,
            "headers": {},
            "params": {}
        }

    def url(self, url: str) -> "RequestBuilder":
        self._options["url"] = url
        return self

    def method(self, method: HttpMethod) -> "RequestBuilder":
        self._options["method"] = method
        return self

    def header(self, key: str, value: str) -> "RequestBuilder":
        self._options["headers"][key] = value
        return self
        
    def headers(self, headers: Dict[str, str]) -> "RequestBuilder":
        self._options["headers"].update(headers)
        return self

    def param(self, key: str, value: Any) -> "RequestBuilder":
        self._options["params"][key] = value
        return self
        
    def params(self, params: Dict[str, Any]) -> "RequestBuilder":
        self._options["params"].update(params)
        return self

    def json(self, data: Any) -> "RequestBuilder":
        self._options["json"] = data
        return self
        
    def data(self, data: Any) -> "RequestBuilder":
        self._options["data"] = data
        return self
        
    def timeout(self, timeout: float) -> "RequestBuilder":
        self._options["timeout"] = timeout
        return self

    def build(self) -> RequestOptions:
        """Get the constructed options."""
        return self._options
